toolkit/components/uniffi-bindgen-gecko-js/src/templates/js/Object.jsm

```diff
@@ -0,0 +1,121 @@
+{%- let object = ci.get_object_definition(name).unwrap() %}
+
+class {{ object.nm() }} {
+    {%- match object.primary_constructor() %}
+    {%- when Some with (cons) %}
+    // Use `init` to instantiate this class.
```

Comment:
The most questionable design choice here is the constructor... the default constructor is just there to assign a pointer to the object and shouldn't be used by consumers. they should use `init` (or alternative constructors if they exist)

This helps us with async as well, since the constructor can't be async (but a static function can)

Reply:
I like this choice.  It seems like the best we can do.